src/osd/OSD.cc

```diff
@@ -4505,6 +4505,16 @@ void OSD::reset_heartbeat_peers(bool all)
       it++;
     }
   }
+
+  // clean up isolated failure info
+  for (map<int,pair<utime_t,entity_inst_t> >::iterator p =
+      failure_pending.begin(); p != failure_pending.end();) {
+
+    if (heartbeat_peers.find(p->first) == heartbeat_peers.end()) {
+      failure_pending.erase(p);
```

Comment:
this looks the wrong way to do the erase and loop simutaneously.

e.g., 
```c++
if (heartbeat_peers.find(p->first) == heartbeat_peers.end()) {
  failure_pending.erase(p++);
} else {
  p++;
}
```

Reply:
done